scripts/cmd/prebuild.js

```diff
@@ -39,11 +39,29 @@ export default async function prebuild(...args) {
 	}
 
 	async function prebuildFile(filepath) {
-		const tscode = await fs.promises.readFile(filepath, 'utf-8');
-		const esbuildresult = await esbuild.transform(tscode, {
-			loader: 'ts',
+		let tscode = await fs.promises.readFile(filepath, 'utf-8');
+		// If we're bundling a client directive, modify the code to match `packages/astro/src/core/client-directive/build.ts`.
+		// If updating this code, make sure to also update that file.
+		if (filepath.includes(`runtime${path.sep}client`)) {
+			tscode = tscode.replace(
+				/export default (.*?)Directive/,
```

Comment:
Can you leave a comment for the regex too? This seems to be an unwritten rule where our directives must be called with a postfix "Directive"; otherwise, this script will fail. Also, they must have an `export default`. We should document it somewhere.

Ideally, this script should throw an error if the regex fails, with a message that tells the developer what went wrong.

Reply:
Yeah I think adding some more guards and comments make sense. I kept it simple since we don't change directive code a lot.